frontend/src/v5/ui/routes/dashboard/projects/federations/federationSettingsForm/federationSettingsForm.component.tsx

```diff
@@ -0,0 +1,318 @@
+					label={formatMessage({
+						id: 'federations.settings.form.unit',
+						defaultMessage: 'Units',
+					})}
+					selectId="unit-label"
+					defaultValue={defaultValues.unit}
+					useFormRegisterProps={register('unit')}
+				>
+					{UNITS.map(({ name, abbreviation }) => (
+						<MenuItem key={abbreviation} value={abbreviation}>
+							{name}
+						</MenuItem>
+					))}
+				</FormSelect>
+				<FormTextField
+					name="code"
+					control={control}
+					label={formatMessage({ id: 'federation.settings.form.code', defaultMessage: 'Code' })}
+					hasError={!!errors.code}
+					helperText={errors.code?.message}
+				/>
+			</FlexContainer>
+			<FormSelectView
+				views={[EMPTY_VIEW].concat(federation.views || [])}
+				federationId={federation._id}
+				defaultValue={defaultValues.defaultView}
+				inputId="default-view-label"
+				label={formatMessage({
+					id: 'federations.settings.form.view',
+					defaultMessage: 'Default View',
+				})}
+				selectId="default-view-label"
+				useFormRegisterProps={register('defaultView')}
+			/>
+			<SectionTitle>GIS servey point</SectionTitle>
+			<FlexContainer>
+				<Controller
+					name="latitude"
+					control={control}
+					render={({ field }) => (
+						<UnitTextField
+							{...field}
+							labelname={formatMessage({ id: 'federations.settings.form.lat', defaultMessage: 'LATITUDE' })}
+							labelunit={formatMessage({ id: 'federations.settings.form.lat.unit', defaultMessage: 'decimal' })}
+							type="number"
+							error={!!errors.latitude}
+							helperText={errors.latitude?.message}
+							required
+						/>
+					)}
+				/>
+				<Controller
+					name="longitude"
+					control={control}
+					render={({ field }) => (
+						<UnitTextField
+							{...field}
+							labelname={formatMessage({ id: 'federations.settings.form.long', defaultMessage: 'LONGITUDE' })}
+							labelunit={formatMessage({ id: 'federations.settings.form.long.unit', defaultMessage: 'decimal' })}
+							type="number"
+							error={!!errors.longitude}
+							helperText={errors.longitude?.message}
+							required
+						/>
+					)}
+				/>
+			</FlexContainer>
+			<Controller
+				name="angleFromNorth"
+				control={control}
+				render={({ field }) => (
+					<UnitTextField
+						{...field}
+						labelname={formatMessage({ id: 'federations.settings.form.angleFromNorth', defaultMessage: 'ANGLE FROM NORTH' })}
+						labelunit={formatMessage({ id: 'federations.settings.form.angleFromNorth.unit', defaultMessage: 'clockwise degrees' })}
+						error={!!errors.angleFromNorth}
+						helperText={errors.angleFromNorth?.message}
+						type="number"
+					/>
+				)}
+			/>
+			<FlexContainer>
+				<Controller
+					name="x"
+					control={control}
+					render={({ field }) => (
+						<UnitTextField
+							labelname="X"
+							labelunit={currentUnit}
+							type="number"
+							error={!!errors.x}
+							helperText={errors.x?.message}
+							required
+							{...field}
+						/>
+					)}
+				/>
```

Comment:
Instead of doing the whole controller thing again, `<UnitTextField />` can be derived from `<FormTextfield />`

Reply:
Fixed. To avoid props leaking I used [Transient props](https://styled-components.com/docs/api#transient-props), let me know if it is not ok and I will change it.